web_src/js/components/DashboardRepoList.vue

```diff
@@ -70,18 +70,16 @@
       </div>
       <div v-if="repos.length" class="ui attached table segment gt-rounded-bottom">
         <ul class="repo-owner-name-list">
-          <li v-for="repo in repos" :key="repo.id">
-            <a class="repo-list-link muted gt-df gt-ac gt-sb" :href="repo.link">
-              <div class="item-name gt-df gt-ac gt-f1">
-                <svg-icon :name="repoIcon(repo)" :size="16" class-name="gt-mr-2"/>
-                <div class="text gt-font-semibold truncate gt-ml-1">{{ repo.full_name }}</div>
-                <span v-if="repo.archived">
-                  <svg-icon name="octicon-archive" :size="16" class-name="gt-ml-2"/>
-                </span>
+          <li class="gt-df gt-ac" v-for="repo in repos" :key="repo.id">
+            <a class="repo-list-link muted gt-df gt-ac gt-f1 gt-min-w-0" :href="repo.link">
```

Comment:
The `repo-list-link` already has its own CSS class name and styles, I think `gt-min-w-0` etc could also be put in the CSS name. Then no need to duplicate the code for another `repo-list-link` below.

ps: why `min-width: 0` is needed?

Reply:
`min-width: 0` is to get text truncation to work.

Reply:
https://css-tricks.com/flexbox-truncated-text/ describes this solution a bit.

Reply:
If it's for a special purpose, I think it can be put into `repo-list-link` class and get some comments.

Or even a special helper class `g-flex-truncate`

Reply:
Don't think a `g-flex-truncate` can be done because the properties need to be on different elements, e.g. `min-width` on the flex child, and the other truncation properties on the element to be truncated, which could be nested arbitrarily deep.

What may work is a wildcard selector, but it may be matching too much.

```
.g-flex-truncate {
  min-width: 0;
}

.g-flex-truncate * {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

Reply:
Moved the style to CSS. Don't feel comfortable adding such a hacky `g-flex-truncate`, so not doing this.